tests/integration/policy/envoy_ratelimit_test.go

```diff
@@ -0,0 +1,196 @@
+	ing         ingress.Instance
+)
+
+func TestRateLimiting_DefaultLessThanOverride(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			res := sendTraffic(ing, t, "Sending traffic...", 300)
+			got429s := float64(res.RetCodes[http.StatusTooManyRequests])
+
+			if got429s <= 0 {
+				t.Errorf("Bad metric value for rate-limited requests (429s): got %f, want at least 1", got429s)
+			}
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		RequireSingleCluster().
+		Label(label.CustomSetup).
+		Setup(istio.Setup(&ist, nil)).
+		Setup(testSetup).
+		Run()
+}
+
+func testSetup(ctx resource.Context) (err error) {
+	bookinfoNs, err = namespace.New(ctx, namespace.Config{
+		Prefix: "istio-bookinfo",
+		Inject: true,
+	})
+	if err != nil {
+		return
+	}
+	if _, err = bookinfo.Deploy(ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookInfo}); err != nil {
+		return
+	}
+
+	ing = ist.IngressFor(ctx.Clusters().Default())
+
+	bookinfoGatewayFile, err := bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespace(bookinfoNs.Name())
+	if err != nil {
+		return
+	}
+	destinationRule, err := bookinfo.GetDestinationRuleConfigFile(ctx)
+	if err != nil {
+		return
+	}
+	destinationRuleFile, err := destinationRule.LoadWithNamespace(bookinfoNs.Name())
+	if err != nil {
+		return
+	}
+	virtualServiceFile, err := bookinfo.NetworkingVirtualServiceAllV1.LoadWithNamespace(bookinfoNs.Name())
+	if err != nil {
+		return
+	}
+	err = ctx.Config().ApplyYAML(bookinfoNs.Name(),
+		bookinfoGatewayFile,
+		destinationRuleFile,
+		virtualServiceFile)
+	if err != nil {
+		return
+	}
+
+	ratelimitNs, err = namespace.New(ctx, namespace.Config{
+		Prefix: "istio-ratelimit",
+	})
+	if err != nil {
+		return
+	}
+
+	err = setupEnvoyFilter(ratelimitNs, ctx)
+	if err != nil {
+		return
+	}
+
+	yamlContent, err := ioutil.ReadFile("testdata/ratelimitservice.yaml")
+	if err != nil {
+		return
+	}
+
+	err = ctx.Config().ApplyYAML(ratelimitNs.Name(),
+		string(yamlContent),
+	)
+	if err != nil {
+		return
+	}
+	// Wait for redis and ratelimit service to be up.
+	fetchFn := kube.NewPodFetch(ctx.Clusters().Default(), ratelimitNs.Name(), "app=redis")
+	if _, err = kube.WaitUntilPodsAreReady(fetchFn); err != nil {
+		return
+	}
+	fetchFn = kube.NewPodFetch(ctx.Clusters().Default(), ratelimitNs.Name(), "app=ratelimit")
+	if _, err = kube.WaitUntilPodsAreReady(fetchFn); err != nil {
+		return
+	}
+
+	// For envoy filter changes to sync.
+	time.Sleep(time.Second * 15)
+
+	return nil
+}
+
+func setupEnvoyFilter(ratelimitNs namespace.Instance, ctx resource.Context) error {
+	content, err := ioutil.ReadFile("testdata/enable_envoy_ratelimit.yaml")
+	if err != nil {
+		return err
+	}
+	con := string(content)
+
+	con = strings.Replace(con, "ratelimit.default.svc.cluster.local",
+		"ratelimit."+ratelimitNs.Name()+".svc.cluster.local", -1)
+
+	ns, err := namespace.Claim(ctx, ist.Settings().SystemNamespace, true)
```

Comment:
Do we need to claim the istio-system namespace here? Since we are only using name of it below, can't we just use `ist.Settings().SystemNamespace` instead? Also if we do need to claim it here, shall we use `ClaimSystemNamespaceOrFail`?

Reply:
Thanks..Fixed